File: android-app/src/main/java/com/medziku/plantwatering/data/parts/SoilMoistureSensorImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.SoilMoistureSensor;


public class SoilMoistureSensorImpl extends SoilMoistureSensor implements Parcelable {

    public SoilMoistureSensorImpl(String name, float soilMoistureLevel) {
        super(name, soilMoistureLevel);
    }

    private SoilMoistureSensorImpl(Parcel in) {
        super(in.readString(), in.readFloat());
    }

    public static final Creator<SoilMoistureSensorImpl> CREATOR = new Creator<SoilMoistureSensorImpl>() {
        @Override
        public SoilMoistureSensorImpl createFromParcel(Parcel in) {
            return new SoilMoistureSensorImpl(in);
        }

        @Override
        public SoilMoistureSensorImpl[] newArray(int size) {
            return new SoilMoistureSensorImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeFloat(getSoilMoistureLevel());
    }
}

File: android-app/src/main/kotlin/com/medziku/plantwatering/data/parts/SoilMoistureSensorImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.SoilMoistureSensor


class SoilMoistureSensorImpl : SoilMoistureSensor, Parcelable {

    constructor(name: String, soilMoistureLevel: Float) : super(name, soilMoistureLevel)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readFloat())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeFloat(soilMoistureLevel)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<SoilMoistureSensorImpl> = object : Parcelable.Creator<SoilMoistureSensorImpl> {
            override fun createFromParcel(`in`: Parcel): SoilMoistureSensorImpl {
                return SoilMoistureSensorImpl(`in`)
            }

            override fun newArray(size: Int): Array<SoilMoistureSensorImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
